CiscoVM/CiscoVM-1.0.0/ciscovm_helpers.py

```diff
@@ -0,0 +1,112 @@
+"""
+Copyright © 2020 Forescout Technologies, Inc.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import logging
+import functools
+import json
+from datetime import datetime, timezone
+
+import requests
+
+
+class CVMHTTPClient:
+    """Cisco Vulnerability Management HTTP client"""
+    CHECK_EVENT_TYPE = "ping"
+    POST_EVENT_TYPE = "job-results"
+
+    def __init__(self, url: str, uid: str, auth_token: str):
+        self.full_url = f"{url.strip('/')}/{uid.strip('/').strip()}"
+        self.auth_token = auth_token.strip()
+
+    def _generate_headers(self, event_type: str = CHECK_EVENT_TYPE) -> dict:
+        """Generate request headers"""
+        return {
+            "X-Forescout-Event": event_type,
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {self.auth_token}",
+        }
+
+    def error_handler(func):
```

Comment:
Is that because Forescout doesn't log exceptions? If that's the case wondering if we should reraise the exception instead of returning False.

Reply:
In ForeScout all exceptions handled and published in logs